api/app/routers/hfi_calc.py

```diff
@@ -0,0 +1,86 @@
+""" Routers for HFI Calculator """
+import logging
+from typing import List
+from fastapi import APIRouter, Response, Depends
+import app
+from app import wildfire_one
+from app.auth import authentication_required, audit
+from app.schemas.hfi_calc import HFIWeatherStationsResponse, WeatherStationProperties,\
+    FuelType, FireCentre, PlanningArea, WeatherStation
+from app.db.crud.hfi_calc import get_fire_weather_stations
+
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(
+    prefix="/hfi-calc",
+    dependencies=[Depends(authentication_required), Depends(audit)]
+)
+
+
+@router.get('/fire-centres', response_model=HFIWeatherStationsResponse)
+# pylint: disable=too-many-locals
+async def get_fire_centres(response: Response):
+    """ Returns list of fire centres and planning area for each fire centre,
+    and weather stations within each planning area. Also returns the assigned fuel type
+    for each weather station. """
+    try:
+        logger.info('/hfi-calc/fire-centres')
+        response.headers["Cache-Control"] = "max-age=0"  # don't let the browser cache this
+
+        with app.db.database.get_read_session_scope() as session:
+            # Fetch all fire weather stations from the database.
+            station_query = get_fire_weather_stations(session)
+            # Prepare a dictionary for storing station info in.
+            station_info_dict = {}
+            # Prepare empty data structures to be used in HFIWeatherStationsResponse
+            fire_centres_list = []
+            planning_areas_by_fire_center = {}
+            stations_by_planning_area = {}
+            # Iterate through all the database records, collecting all the data we need.
+            for (station_record, fuel_type_record, planning_area_record, fire_centre_record) in station_query:
+                fire_centre = FireCentre(name=fire_centre_record.name)
+                planning_area = PlanningArea(name=planning_area_record.name, fire_centre=fire_centre)
+                fire_centres_list.append(fire_centre)
+                planning_areas_by_fire_center.get(fire_centre.name).append(planning_area)
```

Comment:
If planning_areas_by_fire_center.get(fire_centre.name) returns None (likely on the 1'st iteration) this will throw an exception.